### PR TITLE
Updates for 0.12

### PR DESCRIPTION
**Pull request by kRITZCREEK**

### CODE REVIEW

File: src/Data/Codec/Argonaut/Common.purs [hunk 1]
```diff
@@ -80,5 +80,5 @@ map codecA = dimap M.toUnfoldable M.fromFoldable <<< array <<< tuple codecA
 -- | A codec for `StrMap` values.
 -- |
 -- | Encodes as an array of two-element key/value arrays in JSON.
-strMap ∷ ∀ a. JsonCodec a → JsonCodec (SM.StrMap a)
-strMap = dimap SM.toUnfoldable SM.fromFoldable <<< array <<< tuple string
+strMap ∷ ∀ a. JsonCodec a → JsonCodec (FO.Object a)
+strMap = dimap FO.toUnfoldable FO.fromFoldable <<< array <<< tuple string
```
Comment (garyb) [inline src/Data/Codec/Argonaut/Common.purs]: Maybe we should rename this to `object` too?

File: src/Data/Codec/Argonaut/Compat.purs [hunk 2]
```diff
@@ -42,17 +42,17 @@ maybe codec = basicCodec dec enc
 -- | Encodes as a JSON object with the keys as properties.
 -- |
 -- | ```purescript
--- | encode (strMap int) (Data.StrMap.fromFoldable [Tuple "a" 1, Tuple "b" 2]) == "{ \"a\": 1, \"b\": 2}"
+-- | encode (strMap int) (Foreign.Object.fromFoldable [Tuple "a" 1, Tuple "b" 2]) == "{ \"a\": 1, \"b\": 2}"
 -- | ```
-strMap ∷ ∀ a. JsonCodec a → JsonCodec (SM.StrMap a)
+strMap ∷ ∀ a. JsonCodec a → JsonCodec (FO.Object a)
```
Comment (garyb) [inline src/Data/Codec/Argonaut/Compat.purs]: Likewise ☝️

File: test/Test/Migration.purs [hunk 3]
```diff
@@ -92,42 +93,42 @@ propRenameField = do
...
+  → JA.JsonCodec (FO.Object J.Json)
   → Result
 testMigrationCodec { key, expectedValue, input } codec =
   case JA.decode codec (J.fromObject input) of
     Left err → Failed (JA.printJsonDecodeError err)
     Right obj →
-      case SM.lookup key obj of
-        Just value → value === expectedValue
+      case FO.lookup key obj of
+        Just value → J.stringify value === J.stringify expectedValue
```
Comment (garyb) [inline test/Test/Migration.purs]: `Json` isn't `Eq` anymore? Did I mess up somewhere?

File: bower.json [hunk 4]
```diff
@@ -16,15 +16,16 @@
...
   ],
   "dependencies": {
-    "purescript-argonaut-core": "^3.1.1",
-    "purescript-codec": "^2.1.0",
-    "purescript-generics-rep": "^5.4.0",
-    "purescript-variant": "^4.0.0",
-    "purescript-typelevel-prelude": "^2.6.0",
-    "purescript-type-equality": "^2.1.0"
+    "purescript-argonaut-core": "^4.0.0",
+    "purescript-codec": "v3.0.0",
```
Comment (garyb) [inline bower.json]: Humpity bump

File: test/Test/Migration.purs (hunk 3 above)
Comment (kRITZCREEK) [reply test/Test/Migration.purs]: Json isn't Show

Comment (garyb) [reply test/Test/Migration.purs]: Oh, triple equals, right. 👍

File: src/Data/Codec/Argonaut/Common.purs (hunk 1 above)
Comment (kRITZCREEK) [reply src/Data/Codec/Argonaut/Common.purs]: Good catch! I chose foreignObject, since object is already taken

File: bower.json (hunk 4 above)
Comment (kRITZCREEK) [reply bower.json]: Bumped that already :P 3.0.0 is the version you just released ;)

Comment (garyb) [reply bower.json]: 🤦‍♂️

**Pull request merged**